cli/src/Morphir/Web/DevelopApp.elm

```diff
@@ -624,10 +625,51 @@ update msg model =
                     )
 
                 ShowSaveTestError ->
-                    ({model | showSaveTestError = True}, Cmd.none)
+                    ( { model | showSaveTestError = True }, Cmd.none )
 
         ServerGetAttributeResponse attributes ->
-            ( { model | customAttributes = attributes }
+            let
+                -- We also want set the attributeStates once the attributes are fetched
+                attributeEditorState : SDKDict.Dict AttrValueDetail ValueEditor.EditorState
+                attributeEditorState =
+                    attributes
+                        |> Dict.toList
+                        |> List.map
+                            (\( customAttributeId, customAttributeDetail ) ->
+                                let
+                                    selectedModuleName =
+                                        model.homeState.selectedModule
+                                            |> Maybe.map Tuple.second
+                                            |> Maybe.withDefault []
+
+                                    nodeId =
+                                        ModuleID selectedModuleName
+
+                                    attributeValueDetails =
+                                        AttrValueDetail customAttributeId nodeId
+
+                                    irValue =
+                                        customAttributeDetail.data
+                                            |> SDKDict.get nodeId
```

Comment:
As mentioned before types would help a lot but also the naming should reflect what this is. It looks like a `Maybe` so `maybeIRValue` would be a better name for it. Although I would even consider inlining this as well since it's only used once and very small.

Reply:
This has been renamed to maybeIrValue